tools/check_constitution.py

```
@@ -0,0 +1,331 @@
+                    )
+            for num in D_TOKEN.findall(line):
+                if int(num) not in known:
+                    findings.append(
+                        f"citation (§12): {rel}:{n} [D-{num}] — no entry file for it"
+                    )
+
+    # A rule unit added in this diff must cite this pull request's own entry:
+    # a new rule arrives with the decision that made it, or the entry is missing.
+    base_statute = _at(base_sha, STATUTE)
+    if base_statute is not None and pr:
+        was = {l for l in base_statute.split("\n") if RULE_UNIT.match(l)}
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)